core/src/main/resources/db/changelog/changesets/2023_10_03_sendgrid_event.yaml

```diff
@@ -0,0 +1,19 @@
+databaseChangeLog:
+  - changeSet:
+      id: "sendgrid_event"
+      author: mbemis
+      changes:
+        - createTable:
+            tableName: sendgrid_event
+            columns:
+              - column: { name: id, type: uuid, defaultValueComputed: gen_random_uuid(), constraints: { nullable: false, primaryKey: true } }
+              - column: { name: msg_id, type: text, constraints: { nullable: false, unique: true } }
```

Comment:
Is `msg_id` a Sendgrid ID? Is it important to keep? (Do we need to go back to the sendgrid service during the period they retain their info?)

Reply:
I now see why this is important to deal to help us handle updates, since we need to page and deal with date ranges, but I am still curious if `msg_id` is potentially useful in other ways.

Reply:
It'll be potentially useful for tying these activity events with the actual notification events that triggered them from within Juniper. SendGrid returns a partial message ID when you initially send the email, so if we also save that partial ID (not done in this PR), we can then attempt to correlate that with the `msg_id` in this new activity table. This comment has a little bit more context about this: https://github.com/broadinstitute/juniper/pull/576#discussion_r1345033213

I also imagine it's useful to have the `msg_id` if we ever need to follow up with SendGrid support about an issue with a specific email for some reason, especially if it falls outside of our 30 day activity window. They'll only give us 30 days of activity, but I would be very surprised if they didn't hold onto this information for much longer.